pintc/src/predicate/transform/clean.rs

```diff
@@ -23,12 +25,21 @@ pub(crate) fn clean_dead_state_decls(contract: &mut Contract) {
             }
         }
 
+        let mut state_expr: Option<ExprKey> = None;
+
         if let Some(pred) = contract.preds.get_mut(pred_key) {
             for dead_state in dead_state_decls {
+                // if dead_state.get(pred)
+                state_expr = Some(dead_state.get(pred).expr);
+
                 pred.states.remove(dead_state);
             }
         }
 
+        if let Some(expr) = state_expr {
+            println!("expr: {}", contract.with_ctrct(expr));
+        } // TODO: Find a way to deal with the states being mut. For now we are not ignoring them and will remove them if they are unused
```

Comment:
That's fine.. no need to worry about mutability. That's all handled in the pass `lower_storage_accesses`. By the time your pass runs, all of that stuff in consumed.